### PR TITLE
SPLAT-881 - plugin openshift-tests should create junit on failures

### PR DESCRIPTION
**Pull request by mtulio**
https://issues.redhat.com/browse/SPLAT-881

### CODE REVIEW

**Comment by mtulio** [comment]
Converting to draft. The results are unstable when two JUnits are created, due the multiple errors. It should be handled on the main trap function.

Ideas to fix:
- merge JUnit files; 
- if merge will be complex for bash: create a new JUnit naming multiple errors, saving the original files on payloads using the `failure` fields.

```
$ ../openshift-provider-cert-linux-amd64-v0.1.0-devel results 202211242023_sonobuoy_9bb496af-d62d-45be-95e1-f56554ee93d0.tar.gz
Plugin: openshift-conformance-validated
Status: unknown
Total: 1
Passed: 0
Failed: 0
Skipped: 0
unknown: 1

Plugin: openshift-kube-conformance
Status: unknown
Total: 1
Passed: 0
Failed: 0
Skipped: 0
unknown: 1

$ cat results-msgfailed-03/plugins/openshift-kube-conformance/sonobuoy_results.yaml 
name: openshift-kube-conformance
status: unknown
meta:
  type: summary


$ cat results-msgfailed-03/plugins/openshift-kube-conformance/results/global/result 
{"error":"open /tmp/sonobuoy/results/junit_failed_e2e_20221124-202402.xml\njunit_failed_e2e_20221124-202403.xml: no such file or directory"}
```

File: openshift-tests-provider-cert/plugin/global_fn.sh [hunk 1]
```diff
@@ -33,10 +33,22 @@ create_dependencies_plugin() {
 # the intiial values defined on the variables PLUGIN_NAME, CERT_TEST_SUITE,
 # and PLUGIN_BLOCKED_BY.
 init_config() {
-    os_log_info_local "[init_config]"
+    os_log_info_local "[init_config] starting..."
+
+    # Forcing to use newer CLI after plugin renaming
+    if [[ -n "${CERT_LEVEL:-}" ]]
+    then
+        err="[init_config] Wrong CLI version. Please update the openshift-provider-cert binary and try again [CERT_LEVEL=${CERT_LEVEL}]"
```
Comment (mtulio) [inline openshift-tests-provider-cert/plugin/global_fn.sh:41]:
```suggestion
    if [[ -n "${PLUGIN_ID:-}" ]]
    then
        err="[init_config] Wrong CLI version. Please update the openshift-provider-cert binary and try again [PLUGIN_ID=${PLUGIN_ID}]"
```

**Comment by bostrt** [comment]
Thanks for this! I ran into this yesterday when testing on 4.12 and your solution looks good. 

/lgtm
/approve

**Review by bostrt — APPROVED** [review]

**Pull request merged**